From 211675f2f94d5e72883a66f755f3f3c2cfe17e15 Mon Sep 17 00:00:00 2001
From: Marcel Greter <marcel.greter@ocbnet.ch>
Date: Mon, 16 Oct 2017 13:15:54 +0200
Subject: [PATCH 2/3] Improve null terminated lists handling

---
 src/context.cpp | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/src/context.cpp b/src/context.cpp
index 4695e7fd..3725311f 100644
--- a/src/context.cpp
+++ b/src/context.cpp
@@ -101,9 +101,12 @@ namespace Sass {
 
     // load plugins and register custom behaviors
     for(auto plug : plugin_paths) { plugins.load_plugins(plug); }
-    for(auto fn : plugins.get_headers()) { c_headers.push_back(fn); }
-    for(auto fn : plugins.get_importers()) { c_importers.push_back(fn); }
-    for(auto fn : plugins.get_functions()) { c_functions.push_back(fn); }
+    const std::vector<Sass_Importer_Entry> headers = plugins.get_headers();
+    const std::vector<Sass_Importer_Entry> importers = plugins.get_importers();
+    const std::vector<Sass_Function_Entry> functions = plugins.get_functions();
+    for(Sass_Importer_Entry fn : headers) { c_headers.push_back(fn); }
+    for(Sass_Importer_Entry fn : importers) { c_importers.push_back(fn); }
+    for(Sass_Function_Entry fn : functions) { c_functions.push_back(fn); }
 
     // sort the items by priority (lowest first)
     sort (c_headers.begin(), c_headers.end(), sort_importers);
-- 
2.21.0.windows.1

