From 915074356eacf3d80d33d09b6101d67c700f8faf Mon Sep 17 00:00:00 2001
From: Marcel Greter <marcel.greter@ocbnet.ch>
Date: Sat, 18 May 2019 18:47:19 +0200
Subject: [PATCH 09/12] Fix string initialization invocation

---
 src/color_maps.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/color_maps.cpp b/src/color_maps.cpp
index a3de078b..96120645 100644
--- a/src/color_maps.cpp
+++ b/src/color_maps.cpp
@@ -616,7 +616,7 @@ namespace Sass {
   const Color_RGBA* name_to_color(const std::string& key)
   {
     // case insensitive lookup.  See #2462
-    std::string lower{key};
+    std::string lower(key);
     std::transform(lower.begin(), lower.end(), lower.begin(), ::tolower);
 
     auto p = names_to_colors.find(lower.c_str());
-- 
2.21.0.windows.1

