From c08c06a98d68e9ae18e23fb16d2e1334dd2db322 Mon Sep 17 00:00:00 2001
From: Marcel Greter <marcel.greter@ocbnet.ch>
Date: Sun, 19 May 2019 17:24:11 +0200
Subject: [PATCH 11/12] Improve null terminated lists handling

---
 src/context.cpp | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/src/context.cpp b/src/context.cpp
index 73e118db..af268f1c 100644
--- a/src/context.cpp
+++ b/src/context.cpp
@@ -112,11 +112,13 @@ namespace Sass {
     collect_plugin_paths(c_options.plugin_paths);
 
     // load plugins and register custom behaviors
-    for(auto plug : plugin_paths) { plugins.load_plugins(plug); }
-    for(auto fn : plugins.get_headers()) { c_headers.push_back(fn); }
-    for(auto fn : plugins.get_importers()) { c_importers.push_back(fn); }
-    for(auto fn : plugins.get_functions()) { c_functions.push_back(fn); }
-
+    for(auto plug : plugin_paths) { plugins.load_plugins(plug); }
+    const std::vector<Sass_Importer_Entry> headers = plugins.get_headers();
+    const std::vector<Sass_Importer_Entry> importers = plugins.get_importers();
+    const std::vector<Sass_Function_Entry> functions = plugins.get_functions();
+    for(Sass_Importer_Entry fn : headers) { c_headers.push_back(fn); }
+    for(Sass_Importer_Entry fn : importers) { c_importers.push_back(fn); }
+    for(Sass_Function_Entry fn : functions) { c_functions.push_back(fn); }
     // sort the items by priority (lowest first)
     sort (c_headers.begin(), c_headers.end(), sort_importers);
     sort (c_importers.begin(), c_importers.end(), sort_importers);
-- 
2.21.0.windows.1

